androidApp/src/main/java/com/example/tasky/android/agenda/viewmodel/AgendaDetailsViewModel.kt

```diff
@@ -232,9 +233,32 @@ class AgendaDetailsViewModel(
 
     private fun saveUpdate() {
         viewModelScope.launch {
-            agendaRepository.updateAgenda(screenStateFlow.value.agendaItem).onSuccess {
-                _screenStateFlow.update {
-                    it.copy(isEdit = false)
+            val agendaItem = screenStateFlow.value.agendaItem
+
+            when (agendaItem) {
+                is Task -> agendaRepository.updateTask(agendaItem)
+                is Reminder -> agendaRepository.updateReminder(agendaItem)
+                is Event -> {
+                    val eventIsGoing = screenStateFlow.value.eventIsGoing
+                    agendaRepository.updateEvent(
+                        body =
+                            UpdateEventBody(
+                                event = agendaItem,
+                                deletedPhotoKeys = deletedPhotoKeys,
+                                isGoing = eventIsGoing,
+                                photos = emptyList(),
+                            ),
+                    )
+                }
+            }.onSuccess {
+                if (it is Event) {
+                    _screenStateFlow.update { state ->
+                        state.copy(agendaItem = it, isEdit = false)
+                    }
+                } else {
+                    _screenStateFlow.update { state ->
+                        state.copy(isEdit = false)
+                    }
```

Comment:
What about the updates to a task or reminder? Don’t they need to be added to the state?